salsah/src/public/js/jquery.searchbox.js

```diff
@@ -206,7 +208,12 @@
 							left: pos.left,
 							top: pos.top
 						}).show();
-						$this.val('');
+						// if the input already has a value and it is valid, back it up
```

Comment:
great, thx.

> Thanks for the "javascript, the good parts" reference!

It contains some useful information, but some really academic cases too. 

check out the JS pope: https://www.youtube.com/watch?v=NPB34lDZj3E